src/libos/crates/io-uring-callback/src/io_handle.rs

```diff
@@ -44,10 +48,13 @@ impl IoHandle {
     }
 
     /// Cancel the I/O request.
-    ///
-    /// This is NOT implemented, yet.
-    pub fn cancel(&self) {
-        self.0.cancel()
+    pub fn cancel(&self, io_uring: &IoUring) {
+        self.0.cancel(io_uring)
+    }
+
+    /// returns the user_data of the I/O request.
+    pub fn user_data(&self) -> u64 {
+        self.0.user_data()
     }
```

Comment:
Do not expose `user_data` through `IoHandle`. `user_data` has been used by `IoUring` internally. The user of `IoUring` or `IoHandle` should never have to deal with `user_data`.

Reply:
However, io_uring OP, such as, poll_remove, timeout_remove, async_cancel, etc..., need user_data arg.

If we do not expose `user_data` through IoHandle, io-uring-callback can not privide some io-uring operations directly to users. But we can privide these operations in higher level abstraction, like,  `handle.cancel()`